src/components/Clock.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC, useEffect, useState } from 'react';
+
+export function getTimeHelper(): string {
+  return new Date().toUTCString().slice(-12, -4);
+}
+
+type Props = {
+  name: string,
+};
+
+export const Clock: FC<Props> = ({
+  name: clockName,
+}) => {
+  const [today, setToday] = useState(getTimeHelper());
+  const [cachedClockName, setCachedClockName] = useState(clockName);
+
+  useEffect(() => {
+    const timerClockSeconds = window.setInterval(() => {
+      const now = getTimeHelper();
+
+      // eslint-disable-next-line no-console
+      console.info(now);
+      setToday(now);
+    }, 1000);
+
+    return () => {
+      window.clearInterval(timerClockSeconds);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (cachedClockName !== clockName) {
+      setCachedClockName(clockName);
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed to ${clockName}`);
+    }
+  }, [clockName]);
```

Comment:
You don't need that check because this useEffect run if clockName from the dependency array change, so it already keep track and run only if its change. So you also can remove the extra state of cachedClockname

Reply:
Agree 💯! I had it like so initially but somehow it doesn't pass the tests. Tried to work around it and eventually this fixed the errors